src/yoroi-wallets/walletManager.ts

```diff
@@ -348,7 +354,11 @@ export class WalletManager {
   ) {
     const Wallet = this.getWalletImplementation(implementationId)
     const id = uuid.v4()
-    const wallet = new Wallet(storage, networkId, id)
+
+    const apiUrl = getCardanoNetworkConfigById(networkId, provider).BACKEND.API_ROOT
+    const utxoManager = await makeUtxoManager(id, apiUrl, storage)
+
```

Comment:
to build the utxo manager

Reply:
that was my thought when adding it, since it's part of its own initialization maybe add a static method (that will be part of the interface) to prepare the constructor?  (async issue) what do you think?